### ISSUE TITLE
[YSQL] Spurious "Catalog Version Mismatch: A DDL occurred while processing this query"

### ISSUE THREAD

**Issue by bllewell** [YugaByte/yugabyte-db#1457]
I reproduced the issue described here using a single node YugaByte DB cluster created using Version 1.2.9.0. I used a MacBook running macOS Mojave Verison 10.14.5.

The following account provides the text of two `.sql` files. Save them as: (1) `Startup.sql`; and (2) `Catalog_Version_Mismatch.sql` as indicated. The second file uses the first file via the `\i` command. You will need two concurrent `ysqlsh` sessions as explained in comments in the second file. I refer to these as "BLUE" and "YELLOW" because I found it helpful to set the background colors of my two terminal windows differently.

The issue manifests, or does not, according to the precise moment at which you start the second `ysqlsh` session.

Step through `Catalog_Version_Mismatch.sql` by hand, pasting the sections of code into the BLUE or the YELLOW session as explained in the comments.

```
----- Startup.sql ------------------------------------------
-- Improve readability 
\set ECHO 'none'
\set QUIET 'on'
\pset footer off
\set PROMPT1 '%R%  '
\set PROMPT2 '%R%  '

-- Reduce confusion in demos of "isolation level" functionality
\set AUTOCOMMIT 'off'
------------------------------------------------------------
```

```
----- Catalog_Version_Mismatch.sql -------------------------
--
-- ERROR: Catalog Version Mismatch:
-- A DDL occurred while processing this query. Try Again.
--
-- To provoke the error, start ysqlsh in two sessions so that
-- each sees the same schema in the same database and is running
-- as the same user who can create, drop, and populate tables.
-- Then follow the instructions.
--
-- To the avoid the error, start only the BLUE ysqlsh session.
-- Follow the "In the BLUE ysqlsh session" instructions.
-- Only then, start the YELLOW ysqlsh session and continue from
-- "In the YELLOW ysqlsh session"

-- In the BLUE ysqlsh session
\i Startup.sql
begin isolation level read committed;
drop table if exists t;
create table t(n int primary key);
insert into t(n)
select generate_series(0, 9);
commit;

select n from t order by n; rollback;

begin isolation level read committed;
insert into t(n) values(17);
select n from t order by n;

-- In the YELLOW ysqlsh session
\i Startup.sql
begin isolation level read committed;

-- OK so far.
-- But the "insert" causes this:
-- ERROR: Catalog Version Mismatch: A DDL occurred while processing this query. Try Again.
insert into t(n) values(42);

------------------------------------------------------------
```

**Comment by eskuai** [comment]
postgres=# \c postgres
You are now connected to database "postgres" as user "postgres".
postgres=# \d employee
                    Table "public.employee"
  Column  |       Type        | Collation | Nullable | Default
----------+-------------------+-----------+----------+---------
 id       | integer           |           | not null |
 name     | character varying |           |          |
 age      | integer           |           |          |
 language | character varying |           |          |
Indexes:
    "employee_pkey" PRIMARY KEY, lsm (id HASH)

postgres=# select * from employee;
select * from employee;
ERROR:  Query error: Catalog Version Mismatch: A DDL occurred while processing this query. Try Again.
postgres=#

**Comment by louissheng** [comment]
I encountered the issue when using ysqlsh binary from yugabyte-ce-1.2.11.0. Use the ysqlsh in the ~/bin folder in the t-server. Every query fails, even after retry.
ysqlsh (11.2)
Type "help" for help.
postgres=# \d link;
 id          | integer                |           | not null | nextval('link_id_seq1'::regclass)
 url         | character varying(255) |           | not null | 
 name        | character varying(255) |           | not null | 
 description | character varying(255) |           |          | 
 rel         | character varying(50)  |           |          | 
postgres=# select * from link;
ERROR:  Query error: Catalog Version Mismatch: A DDL occurred while processing this query. Try Again.


Initially there's some SQL workloads running when the issue occurs. After stopping the workload, i reconnect ysqlsh, issue still occurs. The workload is 'SqlInserts' from yb-sample-apps(https://github.com/YugaByte/yb-sample-apps).

**Comment by m-iancu** [comment]
Should be fixed now, there were two separate issues fixed in c7b3b0d and ee336d6.
